base/platform/linux/base_file_utilities_linux.cpp

```diff
@@ -196,17 +196,9 @@ QString CurrentExecutablePath(int argc, char *argv[]) {
 		return appimagePath;
 	}
 
-	constexpr auto kMaxPath = 1024;
-	char result[kMaxPath] = { 0 };
-	auto count = readlink("/proc/self/exe", result, kMaxPath);
-	if (count > 0) {
-		auto filename = QFile::decodeName(result);
-		auto deletedPostfix = qstr(" (deleted)");
-		if (filename.endsWith(deletedPostfix)
-			&& !QFileInfo::exists(filename)) {
-			filename.chop(deletedPostfix.size());
-		}
-		return filename;
+	const auto exeLink = QFileInfo(u"/proc/%1/exe"_q.arg(getpid()));
```

Comment:
That doesn't cut it since FreeBSD and DragonFly use `file` not `exe`.

This is not an improvement, imho.

Reply:
They don't have procfs enabled by default anyway

Reply:
> They don't have procfs enabled by default anyway

You are mistaken.

NetBSD enables procfs(5) by default just like Linux, as I already stated in
https://github.com/desktop-app/lib_base/pull/114#issuecomment-1060408957.

Reply:
But you talked about FreeBSD in previous message, not NetBSD...

Reply:
Admittedly, this diff would work on Linux and NetBSD now, but it would still be incorrect for the rest.
Do you want to fix stuff piece by piece whenever some OS changes their defaults?

I don't see much value in this approach, why not just fix it properly as I suggested, regardless of what the default behaviour is today?

Reply:
Does FreeBSD plan to change their default? If not, then IMHO it's better to have a sysctl-based solution or AT_EXECPATH like Qt does

Reply:
> Does FreeBSD plan to change their default?

No idea.

> If not, then IMHO it's better to have a sysctl-based solution or AT_EXECPATH-based one like Qt does

We can always do that in a follow-up commit once someone on FreeBSD chimes in (I don't use it).

Reply:
I just double checked by installing DragonFly and they also enable procfs by default.
Is that enough reason to take the proper OS aware approach in #114 instead?

Reply:
Wouldn't it be better to handle that with sysctl/AT_EXECPATH since it should work on all FreeBSD derivatives?

Reply:
> Wouldn't it be better to handle that with sysctl/AT_EXECPATH since it should work on all FreeBSD derivatives?

Probably, but that needs proper testing on those OSes.
For now, the PATH lookup should do on all of them.